### PR TITLE
Update rerank_server_demo.py

### PR DESCRIPTION
**Pull request by zsuroy**
# Pull Request

## 描述 (Description)
1. Added CUDA debugging support
2. Improved memory management during model inference.
3. Fixed logits output

## 变更类型 (Type of Change)

- [x] 🐛 Bug 修复 (Bug fix)
- [x] ⚡ 性能优化 (Performance improvement)

## 影响范围 (Scope)

- [x] 其他 (Other): Python rerank demo

## 测试 (Testing)

- [x] 集成测试 (Integration tests)
- [x] 手动测试 (Manual testing)
- [x] API 测试 (API testing)

## 检查清单 (Checklist)

- [x] 代码遵循项目的编码规范
- [x] 已进行自我代码审查
- [x] 代码变更已添加适当的注释
- [x] 相关文档已更新
- [x] 变更不会产生新的警告

## 其他信息 (Additional Information)

Test Device: 
Windows WSL Ubuntu 20.04 (python3.10 CUDA 5090 32G)

### CODE REVIEW

**Comment by zsuroy** [comment]
Tested models：bge-reranker-base, bge-reranker-v2-m3(Good!), bge-reranker-v2-gamma
